modules/fedimint-ln-common/src/api.rs

```diff
@@ -220,6 +290,16 @@ where
     }
 }
 
+/// Request sent to the federation that requests the removal of a gateway
+/// registration. Each peer is expected to check the `signatures` map for the
+/// signature that validates the gateway authorized the removal of this
+/// registration.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct RemoveGatewayRequest {
+    pub gateway_id: PublicKey,
+    pub signatures: BTreeMap<PeerId, Signature>,
```

Comment:
Not ideal that we have to send all the signatures to all guardians, but not something worth spending time on optimizing.

Reply:
Agreed :/